File: abc.com/src/main/java/application/auth/roles/Role.java
package application.auth.roles;

import org.springframework.data.annotation.Id;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by gipai on 9/28/2017.
 */
public class Role {

    @Id
    private String id;
    private String name;


    private List<PermissionType> permissions = new ArrayList<PermissionType>();

    public Role(){

    }

    public Role(String name,List<PermissionType> permissions){
        this.name = name;
        this.permissions = permissions;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public List<PermissionType> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<PermissionType> permissions) {
        this.permissions = permissions;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean addPermission(PermissionType permission) {
        int length = permissions.size();
        for(int i=0; i < length; i++){
            if(permissions.get(i) == permission){
                //return as already exists
                return false;
            }
        }
        permissions.add(permission);
        return true;
    }

    public boolean deletePermission(PermissionType permission) {
        int length = permissions.size();
        for(int i=0; i < length; i++){
            if(permissions.get(i) == permission){
                permissions.remove(i);
                return true;
            }
        }
        return false;
    }



}

File: abc.com/digi-client1/src/app/users/role.ts
/**
 * Created by gipai on 10/3/2017.
 */
import { Class } from '@angular/core';



export class Role {

  public  id: string;
  public  name: string;
  public  permissions: Array<string>;


  constructor() {
  }


  public setName ( name: string ) {
    this.name = name;
  }

  public getName(): string {
    return this.name;
  }

  public setId ( id: string ) {
    this.id = id;
  }
  public getId (): string {
    return this.id ;
  }

  public setPermissions ( permissions: Array<string> ) {
    this.permissions = permissions;
  }
  public getPermissions (): Array<string> {
    return this.permissions ;
  }
}
